streams/src/main/java/org/apache/kafka/streams/processor/TaskMetadata.java

```diff
@@ -32,10 +34,22 @@
 
     private final Set<TopicPartition> topicPartitions;
 
+    private final Map<TopicPartition, Long> committedOffsets;
+
+    private final Map<TopicPartition, Long> endOffsets;
+
+    private Optional<Long> timeCurrentIdlingStarted;
+
     public TaskMetadata(final String taskId,
-                        final Set<TopicPartition> topicPartitions) {
+                        final Set<TopicPartition> topicPartitions,
+                        final Map<TopicPartition, Long> committedOffsets,
+                        final Map<TopicPartition, Long> endOffsets,
+                        final Optional<Long> timeCurrentIdlingStarted) {
         this.taskId = taskId;
         this.topicPartitions = Collections.unmodifiableSet(topicPartitions);
+        this.committedOffsets = committedOffsets;
+        this.endOffsets = endOffsets;
```

Comment:
Should these be wrapped with `Collections.unmodifiableMap()` or `ImmutableMap.copyOf()`? If the two maps are not expected to be modified after passing them as parameter, then I suggest `ImmutableMap.copyOf()`.

Reply:
The Metadata map could be immutable I will make this change.